tests/training/metrics_test.py

```diff
@@ -211,3 +213,88 @@ def test_f1_measure_works_for_sequences(self):
         numpy.testing.assert_almost_equal(precision, 1.0)
         numpy.testing.assert_almost_equal(recall, 0.5)
         numpy.testing.assert_almost_equal(f1, 0.66666666666)
+
+
+class SpanBasedF1Test(AllenNlpTestCase):
+
+    def setUp(self):
+        super(SpanBasedF1Test, self).setUp()
+        vocab = Vocabulary()
+        vocab.add_token_to_namespace("O", "tags")
+        vocab.add_token_to_namespace("B-ARG1", "tags")
+        vocab.add_token_to_namespace("I-ARG1", "tags")
+        vocab.add_token_to_namespace("B-ARG2", "tags")
+        vocab.add_token_to_namespace("I-ARG2", "tags")
+        self.vocab = vocab
+
+    def test_conll_span_based_f1_extracts_correct_spans(self):
+        tag_vocab = self.vocab.get_index_to_token_vocabulary("tags")
+        metric = SpanBasedF1Measure(tag_vocab, ignore_classes=["O"])
```

Comment:
There should be a case here for when an actual span label is ignored, and again ignoring "O" does nothing now.